tensorflow/examples/tutorials/mnist/mnist_rnn.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+""" Recurrent Neural Network.
+A Recurrent Neural Network (LSTM) implementation example using TensorFlow library.
+This example is using the MNIST database of handwritten digits (http://yann.lecun.com/exdb/mnist/).
+Links:
+    [Long Short Term Memory](http://deeplearning.cs.cmu.edu/pdfs/Hochreiter97_lstm.pdf).
+    [MNIST Dataset](http://yann.lecun.com/exdb/mnist/).
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.contrib import rnn
+from tensorflow.examples.tutorials.mnist import input_data
+
+def rnn_network(x_in, weight_set, bias_set):
+    """To classify images using a recurrent neural network, we consider every image
+    row as a sequence of pixels. Because MNIST image shape is 28*28px, we will then
+    handle 28 sequences of 28 steps for every sample.
+    Args:
+        x_in: input images.
+        weight_set: collection of input/ output weight.
+        bias_set: collection input/ output bias.
+    Returns:
+        Output tensor with the computed logits.
+    """
+
+    # Prepare data shape to match `rnn` function requirements.
+    # Current data input shape: (BATCH_SIZE, num_step, n_input).
+    # Required shape: 'num_step' tensors list of shape (BATCH_SIZE, n_input).
+    x_in = tf.reshape(x_in, [-1, NUM_INPUT])
+
+    # Hidden layer ==> (128 batch, 28 steps, 128 hidden).
+    x_hidden = tf.matmul(x_in, weight_set['in']) + bias_set['in']
+    x_hidden = tf.reshape(x_hidden, [-1, NUM_STEP, NUM_HIDDEN_UNITS])
+
+    # Define a lstm cell with tensorflow.
+    lstm_cell = rnn.BasicLSTMCell(NUM_HIDDEN_UNITS, forget_bias=1.0, state_is_tuple=True)
+
+    # init zero state, lstm cell is consist of twp parts: (c_state, h_state).
+    init_state = lstm_cell.zero_state(BATCH_SIZE, dtype=tf.float32)
+
+    # Get lstm cell output.
+    outputs, states = tf.nn.dynamic_rnn(lstm_cell, x_hidden, \
```

Comment:
no need for '\'

Reply:
I assume you mean the quota in the comments, right? If so, fixed it~

Reply:
Sorry it didn't print correctly. I meant the backslash `\`.